tests/scanners/test.binary.js

```diff
@@ -3,29 +3,25 @@ import { FLAGGED_FILE_MAGIC_NUMBERS } from 'const';
 import * as messages from 'messages';
 
 describe('Binary', () => {
-  it('should do nothing on a text file', () => {
+  it('should do nothing on a text file', async () => {
     const buffer = Buffer.from('wat');
-    const scanner = new BinaryScanner(buffer, 'wat.txt');
+    const binaryScanner = new BinaryScanner(buffer, 'wat.txt');
 
-    return scanner.scan()
-      .then(({ linterMessages }) => {
-        expect(linterMessages.length).toEqual(0);
-      });
+    const { linterMessages } = await binaryScanner.scan();
+    expect(linterMessages.length).toEqual(0);
   });
 
-  it('should notice a flagged file', () => {
-    FLAGGED_FILE_MAGIC_NUMBERS.forEach((magic) => {
+  it('should notice a flagged file', async () => {
+    FLAGGED_FILE_MAGIC_NUMBERS.forEach(async (magic) => {
```

Comment:
@hameleonka it looks that this test is not going to fail when it should (and this problems seems to be also present in the previous version of this test) because it is not "awaiting" nor "returning" any of the promises (which if rejected they are going to be logged on the console as unhandled rejections, but they will not make the test to fail as expected).

To fix the issue we should:

- use `FLAGGED_FILE_MAGIC_NUMBERS.map` instead of `FLAGGED_FILE_MAGIC_NUMBERS.forEach` (so that we collect all the promises returned by calling the async function passed to the map on all the array elements)
- await (or return) Promise.all(/* result of the FLAGGED_FILE_MAGIC_NUMBERS.map */), so that the test will actually fail when the async function returns a rejection for one of the FLAGGED_FILE_MAGIC_NUMBERS elements)

e.g. it could become something like:

```js
it('should notice a flagged file', async () => {
    await Promise.all(FLAGGED_FILE_MAGIC_NUMBERS.map(async (magic) => {
      const buffer = Buffer.from(magic);
      const binaryScanner = new BinaryScanner(buffer, 'wat.txt');

      const { linterMessages } = await binaryScanner.scan();

      expect(linterMessages.length).toEqual(1);
      expect(linterMessages[0].code).toEqual(
        messages.FLAGGED_FILE_TYPE.code
      );
      expect(linterMessages[0].file).toEqual('wat.txt');
    }));
  });
```

Side notes: if you want to manually verify the behavior of your previous version of this test and the new proposed one, you can change one of the assertions to force it to fail and check that the test is also failing as expected, e.g.:
by changing `expect(linterMessages[0].file).toEqual('wat.txt');` into `expect(linterMessages[0].file).toEqual('non-existent-file.txt');`.